packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -325,26 +327,45 @@ async function commitCache(
 export async function saveCache(
```

Comment:
This has been adapted to choose which upload method we want to use. The legacy version uses a bespoke HTTP client to upload to AC. We want to use the Azure SDK to upload to ABS directly, hence backward compatible change.